simulator/bpu/bpentry.h

```diff
@@ -152,17 +147,15 @@ class BPEntryTwoBit : public BPEntry
     void update( bool is_taken, Addr target)
     {
         if ( is_taken && _target != target)
-        {
-            state.reset();
-            _target = target;
-        }
+            /* if the address has somehow changed, we should update it appropriately */
+            reset( target);
```

Comment:
Previous code was better.

Reply:
Why? When target changes, the entry should be completely reset and updated with new address. Why should I explicitly do a reset here while having already method `reset()`. What if it was not just about resetting the state, but about resetting dozens of internal strcutures?

Reply:
It is expected that `reset()` method changes the state to the default one. You have to add `update_target()` or something like that.

Reply:
I was inspired by something like `std::unique_ptr::reset`. Should I leave reset() without arguments then, and each time the `reset` is called, call `update_target` as well?

Reply:
Yes, please revert `reset()` and add `update_target` instead.

Reply:
No problem, [done](https://github.com/MIPT-ILab/mipt-mips/pull/186/commits/d8e72fd2779e8432ef09382a5c4ae67af4720892).